starknet-foundry/crates/forge/tests/integration/starknet.rs

```diff
@@ -0,0 +1,432 @@
+
+                assert(next_block_number >= block_number, 'block number decrese');
+                assert(next_block_timestamp >= block_timestamp, 'block timestamp decrese');
+                assert(sequencer_address == next_sequencer_address, 'sequencer changed');
+            }
+    "#
+        ),
+        Contract::new(
+            "Blocker",
+            indoc!(
+                r#"
+                use starknet::ContractAddress;
+
+                #[starknet::interface]
+                trait IBlocker<TContractState> {
+                    fn write_block(ref self: TContractState);
+                    fn read_block_number(self: @TContractState) -> u64;
+                    fn read_block_timestamp(self: @TContractState) -> u64;
+                    fn read_sequencer_address(self: @TContractState) -> ContractAddress;
+                }
+
+                #[starknet::contract]
+                mod Blocker {
+                    use array::ArrayTrait;
+                    use starknet::get_block_info;
+                    use box::BoxTrait;
+                    use starknet::ContractAddress;
+
+                    #[storage]
+                    struct Storage {
+                        block_number: u64,
+                        block_timestamp: u64,
+                        sequencer_address: ContractAddress,
+                    }
+
+                    #[external(v0)]
+                    impl IBlockerImpl of super::IBlocker<ContractState> {
+                        fn write_block(ref self: ContractState) {
+                            let block_info = get_block_info().unbox();
+                            self.block_number.write(block_info.block_number);
+                            self.block_timestamp.write(block_info.block_timestamp);
+                            self.sequencer_address.write(block_info.sequencer_address);
+                        }
+
+                        fn read_block_number(self: @ContractState) -> u64 {
+                            self.block_number.read()
+                        }
+                        fn read_block_timestamp(self: @ContractState) -> u64 {
+                            self.block_timestamp.read()
+                        }
+                        fn read_sequencer_address(self: @ContractState) -> ContractAddress {
+                            self.sequencer_address.read()
+                        }
+                    }
+                }
+    "#
+            )
+        )
+    );
+
+    let result = run(
+        &test.path().unwrap(),
+        &test.path().unwrap().join("src/lib.cairo"),
+        &Some(test.linked_libraries()),
+        &Default::default(),
+        Some(&Utf8PathBuf::from_path_buf(corelib().to_path_buf()).unwrap()),
+        &test.contracts(corelib().path()).unwrap(),
+        &Utf8PathBuf::from_path_buf(predeployed_contracts().to_path_buf()).unwrap(),
+    )
+    .unwrap();
+
+    assert_passed!(result);
+}
+
+// TODO Make nonce behavior consistent with Starknet
```

Comment:
Can we link this issue here?
https://github.com/foundry-rs/starknet-foundry/issues/258

Reply:
Creating #292 and linking it. #258 is cheatable starknet related.